src/librustc/dep_graph/dep_node.rs

```diff
@@ -127,6 +127,11 @@ pub enum DepNode<D: Clone + Debug> {
     TraitItems(D),
     ReprHints(D),
     TraitSelect(Vec<D>),
```

Comment:
I did it that way because using `SmallVec` would likely increase the size of `DepNode`. But I haven't measured and I'm not sure how much that would matter.